### PR TITLE
chore(repo): update release notification steps

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## What is the current behavior?

we have a few out of date steps in our release guide

GitHub Issue Number: N/A

## What is the new behavior?

remove the #whathappened Slack channel instructions from the release steps. this channel no longer exists and a new automated channel has been stood up in its place (#github_releases). since there was one item under the 'left folks in Ionic know...' section, remove the section 'header' and just replace it with the discord/blog post stuff.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

N/A - markdown only

### CODE REVIEW

**Review by christian-bromann — APPROVED** [review]
👍

File: RELEASE.md [hunk 1]
```diff
@@ -88,10 +88,5 @@ manual release was performed.
...
       vNUMBER release](https://github.com/ionic-team/stencil/releases/tag/TAG). 
       ```
-1. Let folks in Ionic know about the release:
-   1. Add a message to the `#whathappened` Slack channel with a link to the changelog.
-      1. Let the CS team know by 'at-ing' them  about any enterprise reported issues or feature requests that were included in the release in a thread on your #whathappened post. Jira
-         tickets related to enterprise requests should be marked with the "jira_escalated" label and/or the "Enterprise Support"
-         or "Enterprise Feature Requests" epic.
-   1. If there's a blog post to go out (either today or this week), let the folks in the `#devrel-stencil` channel know about the release and that the blog can go out.
-      1. When the blog goes out, put an announcement in the `#announcements` channel in Discord.
+1. If there's a blog post to go out (either today or this week), let the folks in the `#devrel-stencil` channel know about the release and that the blog can go out.
```
Comment (tanner-reits) [inline RELEASE.md:91]: Is there a new `devrel-stencil` channel?

Comment (rwaskiewicz) [reply RELEASE.md:91]: Oh, good point! We're moving to a more general ask devrel channel now - let me update that real quick

Comment (rwaskiewicz) [reply RELEASE.md]: Done in d0a2b3291aa20c688c1413466c646da8cb305f74

**Pull request merged**